### ISSUE TITLE
Make quality control plots a standard part of mesh generation for E3SM

### ISSUE THREAD

**Issue by xylar** [MPAS-Dev/compass#96]
As discussed [here](https://github.com/MPAS-Dev/compass/pull/62#issuecomment-829788675), it would be really helpful to have quality control plots (like [here](https://github.com/MPAS-Dev/MPAS-Model/pull/555#issuecomment-636912023)) generated automatically as part of creating meshes and initial conditions for E3SM.

This would presumably involve porting Matlab code to python as part of a new test case for the global ocean.

**Comment by xylar** [comment]
@proteanplanet, as a first step toward assessing how much work this would be, would you be willing to point me to your Matlab scripts so I can see how much code we're talking about and how much beyond my rusty Matlab skills this might be?

**Comment by xylar** [comment]
cc @vanroekel

**Comment by dengwirda** [comment]
Hi all, there are also a bunch of mesh quality routines available direct in `jigsaw-python`...

**Comment by xylar** [comment]
Thanks @dengwirda.  Those would also be helpful to include.  I think what is most critical in this context are some things about quality of coast lines related to specific features that @proteanplanet has picked out.  So what I have in mind here are more quality control from a geopolitical and physical oceanography standpoint.  I'm guessing what jigsaw-python provides are metrics that are more related to numerical method?  If so, that would still be a useful addition.  If you have specific suggestions or examples you'd like to point us to, please do!

**Comment by dengwirda** [comment]
@xylar, yes totally agree --- @proteanplanet's coastline/channel quality control is very important on it's own (and makes very nice plots!). I think it could be of interest to add additional metrics though, to ensure geometric quality is consistent with what `TRSK` is happy to deal with. [These](https://github.com/dengwirda/jigsaw-geo-python/blob/24fc1aaf72d7693ae7155e292c38daff03043a1e/tests/case_1_.py#L70) are some standard things that I run in the `jigsaw` examples, with a little more detail [here](https://github.com/dengwirda/jigsaw-python/blob/master/jigsawpy/tools/scorecard.py).

**Comment by xylar** [comment]
@dengwirda, got it.  So these wouldn't be images saved out or fields added to the mesh necessarily but rather output to the log file when the mesh is being created, right?

**Comment by dengwirda** [comment]
@xylar, I think there could be a few possibilities. Printing to the log-file makes sense, but another option could be to use the metrics to identify additional cells/regions (that fall below some quality thresholds, say) to be passed to @proteanplanet's workflow for additional visual inspection?

**Comment by xylar** [comment]
Okay, so for that purpose, we would want to add these metrics to the mesh and pass them through at least to the culling stage.

**Comment by dengwirda** [comment]
@xylar, that would definitely work, but I'm not sure what the easiest path might be at this stage --- another option is that the MPAS mesh data-structure may (I think) have enough information in it to just compute these metrics on the fly within/before @proteanplanet's workflow. 
Does it make sense for the three of us to discuss a little more first, and try to hash out a design?

**Comment by xylar** [comment]
The rest of this work is likely to be substantial.  @proteanplanet's plotting scripts are, to the best of my knowledge, in Matlab.  So porting them to Python will be a not insignificant task.  I'm sure there will be time to discuss adding these metrics to that workflow.  It likely won't happen in the immediate term, though.

**Comment by xylar** [comment]
@proteanplanet, could you point us to your Matlab code for making your quality control plots if it is publicly available?
